lua/lspconfig/configs.lua

```diff
@@ -78,22 +78,29 @@ function configs.__newindex(t, config_name, config_def)
 
     function M.launch()
       local root_dir = get_root_dir(api.nvim_buf_get_name(0), api.nvim_get_current_buf())
-      if not root_dir then
-        vim.notify(string.format('Launching %s failed: matching root directory not detected.', config_name))
-        return
-      end
-      api.nvim_command(
-        string.format(
-          "autocmd %s unsilent lua require'lspconfig'[%q].manager.try_add_wrapper()",
-          'BufReadPost ' .. root_dir .. '/*',
-          config.name
+      if root_dir then
+        api.nvim_command(
+          string.format(
+            "autocmd %s unsilent lua require'lspconfig'[%q].manager.try_add_wrapper()",
+            'BufReadPost ' .. root_dir .. '/*',
+            config.name
+          )
         )
-      )
-      for _, bufnr in ipairs(vim.api.nvim_list_bufs()) do
-        local buf_dir = api.nvim_buf_get_name(bufnr)
-        if buf_dir:sub(1, root_dir:len()) == root_dir then
-          M.manager.try_add_wrapper(bufnr)
+        for _, bufnr in ipairs(vim.api.nvim_list_bufs()) do
+          local buf_dir = api.nvim_buf_get_name(bufnr)
+          if buf_dir:sub(1, root_dir:len()) == root_dir then
+            M.manager.try_add_wrapper(bufnr)
+          end
         end
+      elseif not root_dir and config.single_file_support then
+        -- This allows on_new_config to use the parent directory of the file
+        -- Effectively this is the root from lspconfig's perspective, as we use
+        -- this to attach additional files in the same parent folder to the same server.
+        -- We just no longer send rootDirectory or workspaceFolders during initialization.
+        local pseudo_root = util.path.dirname(api.nvim_buf_get_name(0))
+        M.manager.add(pseudo_root, true)
```

Comment:
Can you not have multiple buffers for the same file, with splits or something? I feel like that isn't covered here.

Reply:
did you try it :)

Reply:
I did and it seems to work fine 🚀

Reply:
I know ;)